presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java

```diff
@@ -150,6 +175,18 @@ public static Object subscript(boolean legacyMissingKey, FunctionInvoker functio
     @UsedByGeneratedCode
     public static Object subscript(boolean legacyMissingKey, FunctionInvoker functionInvoker, MethodHandle keyEqualsMethod, Type keyType, Type valueType, ConnectorSession session, Block map, double key)
     {
+        if (map instanceof MapElementBlock) {
+            MapElementBlock mapBlock = (MapElementBlock) map;
+            int valuePosition = mapBlock.seekKeyExact(key);
+            if (valuePosition == -1) {
```

Comment:
I don't think this is a good idea given the number of arguments (8) that function will have to take.